### DISCUSSION TITLE
Do you know when trending repositories on GitHub are usually updated?

### DISCUSSION THREAD

**Discussion by alidehkhodaei** [community/community#64295] [Other Feature Feedback, Questions, & Ideas] [answered]
I've created a <a href="https://github.com/alidehkhodaei/trending-repositories">bot</a> that fetches <a href="https://github.com/trending"> trending repositories</a>, and I want to determine when I should fetch them to get the most recently updated ones. I believe they are updated every day at a specific time, and I'm trying to find out the exact timing.
I appreciate your help in advance.

**Comment by messiisgreat** [comment]
Sourcegraph: According to the Sourcegraph documentation[1](https://docs.sourcegraph.com/admin/repo/update_frequency), repositories are updated based on a smart heuristic that determines the frequency of past commits in the repository. The frequency at which Sourcegraph polls the code host for updates is determined by this heuristic. For example, if a repository's last commit was 8 hours ago, then the next sync will be scheduled 4 hours from now. If after 4 hours, there are still no new commits, then the next sync will be scheduled 6 hours from then. Repositories will never be updated more frequently than 45 seconds, and no less frequently than every 8 hours.

GitHub: According to a GitHub issue[2](https://github.com/google/go-github/issues/821), you can use the following code to get the last updated time of a repository's source code:

r, _, err := c.Repositories.Get(context.Background(), user, repo)
srcLastUpdated := r.UpdatedAt

Git: According to a Stack Overflow post[3](https://stackoverflow.com/questions/16683433/git-repo-update-date-and-time), you can use the following command to get the date and time when your Git repository was updated from the remote repo:

git log -1 --format="%cd" --date=local

Pluralsight: According to the Pluralsight Help Center[4](https://help.pluralsight.com/help/repository-processing-statuses), the last updated column for a repository can take anywhere from a few minutes to a few hours depending on the size and contents

Bitbucket: According to the Bitbucket Server Knowledge Base[6](https://confluence.atlassian.com/bitbucketserverkb/mirror-takes-a-long-time-to-update-repositories-1168850454.html), mirror updates for multiple repositories may take up to 30 minutes, while other mirrors may update repositories almost instantaneously.

**Reply by alidehkhodaei** [reply]
@skyninjadev I think you misunderstood what I meant.

**Comment by messiisgreat** [comment]
By using these methods, you can determine the timing of when repositories are updated and fetch the most recently updated ones.

**Comment by thelllmike** [comment]
GitHub's daily updates for trending repositories typically occur around 10:00 AM UTC.

**Reply by alidehkhodaei** [reply]
@thelllmike Thank you, I will check it.

**Comment by alidehkhodaei** [comment, marked as answer]
I checked it for two days and found out it was around `11 AM UTC`.

**Reply by lovestaco** [reply]
Source?

**Comment by paradise321** [comment]
Sounds like the time is changed today, at least it's **before** 11 AM UTC
